ext/tiktoken_ruby/extconf.rb

```diff
@@ -1,6 +1,27 @@
 # frozen_string_literal: true
 
-require "mkmf"
-require "rb_sys/mkmf"
+require 'mkmf'
+require 'rb_sys/mkmf'
 
-create_rust_makefile("tiktoken_ruby/tiktoken_ruby")
+# Detect the operating system
+os = RbConfig::CONFIG['host_os']
+
+# Custom configurations for OpenBSD and FreeBSD
+if os.include?('openbsd') || os.include?('freebsd')
+  $CFLAGS << ' -I/usr/local/include'  # Include path for headers
+  ENV['RUSTFLAGS'] = '-C link-arg=-lpthread'  # Link with pthread
```

Comment:
Is there a reason to use the ENV variable rather than `r.extra_rustflags` like in the [rb-sys readme](https://github.com/oxidize-rb/rb-sys/blob/main/gem/README.md)